content/docs/use-cases/ci-and-ml.md

```diff
@@ -0,0 +1,172 @@
+project is very involved and complex. CI is used to perform the testing; let it
+be on various platforms and environments, or simply move the testing load from a
+developer’s local machine to a dedicated testing platform. Just like CI helps
+when it comes to runing complex test suite, it can also help when the need for
+training arises.
+
+![](/img/ci_for_ml_retrain_model.jpg) _Model training flow_
+
+Assuming that in a project’s repository a DVC pipeline is defined for training a
+model. You can ask your CI tool to call `dvc pull` and, voilà, it now has access
+to the _right_ data needed for training. The reproducing pipeline can be
+arbitrary complicated and comprise of many steps; regardless of all, `dvc repro`
+will yield a newly trained model. This new model will be tracked and pushed to
+the remote cache (note `dvc push`) and will be "linked" to the PR.
+
+```yml
+# .github/workflows/train_model.yml
+
+name: Retrain model
+on:
+  pull_request:
+    types: synchronize
+
+jobs:
+  train:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Set up the environment on the runner
+        run: |
+          apt-get update -y
+          apt-get install python3-dev -y
+          pip install -r requirements.txt
+
+      - name: DVC fetch the data used by the branch
+        run: dvc pull
+
+      - name: DVC retrain the model
+        run: dvc repro
+
+      - name: Track the newly trained model
+        run: |
+          dvc add model.pkl
+          git config --global user.name 'Your Name'
+          git config --global user.email 'my@enail.com'
+          git add model.pkl.dvc
+          git commit -m "Model retrained upon sync to PR"
+          git push
+          dvc push
+```
+
+**Bonus (⚡️):** It might be that the training needs a tailored environment.
+This can be achieved by defining a runner that accommodates the required
+environment (for example a GPU).
+
+## Seamless model evaluation
+
+![](/img/ci_for_ml_evaluate_models.jpg) _Evaluating different models against the
+same validation dataset_
+
+An fundamental part of a prediction model’s quality, except its underlying code
+that can and should be tested, is its performance. Just like you wouldn’t merge
+code that fails tests into your `master` branch, you would not want to adopt a
+model that under-performs in comparison to the one you already have deemed
+stable on the `master` branch. It is common that one or more developers are
+maintaining or trying to improve a model that’s considered stable. In this case,
+before replacing this model with a new one, you would like to evaluate the new
+model’s performance and compare it to the performance of the “stable” model. For
+this comparison to be meaningful, you would like to evaluate the performances
+against the _same_ validation set.
+
+```yml
+name: Evaluate model
+on:
+  pull_request:
+    types: synchronize
+
+jobs:
+  evaluate:
+    runs-on: ubuntu-latest
```

Comment:
Same as above.

Reply:
Let's resolve it together with the other one